cnd/src/db/schema.rs

```diff
@@ -1,106 +1,181 @@
+        ethereum_expiry -> BigInt,
+        bitcoin_expiry -> BigInt,
+        secret_hash -> Text,
+    }
 }
 
 table! {
-   rfc003_ethereum_bitcoin_accept_messages {
-       id -> Integer,
-       swap_id -> Text,
-       ethereum_redeem_identity -> Text,
-       bitcoin_refund_identity -> Text,
-       at -> Timestamp,
-   }
+    rfc003_ethereum_bitcoin_accept_messages {
+        id -> Integer,
+        swap_id -> Text,
+        ethereum_redeem_identity -> Text,
+        bitcoin_refund_identity -> Text,
+        at -> Timestamp,
+    }
 }
 
 table! {
-   rfc003_bitcoin_ethereum_accept_messages {
-       id -> Integer,
-       swap_id -> Text,
-       bitcoin_redeem_identity -> Text,
-       ethereum_refund_identity -> Text,
-       at -> Timestamp,
-   }
+    rfc003_bitcoin_ethereum_accept_messages {
+        id -> Integer,
+        swap_id -> Text,
+        bitcoin_redeem_identity -> Text,
+        ethereum_refund_identity -> Text,
+        at -> Timestamp,
+    }
 }
 
 table! {
-   rfc003_decline_messages {
-       id -> Integer,
-       swap_id -> Text,
-       reason -> Nullable<Text>,
-   }
+    rfc003_decline_messages {
+        id -> Integer,
+        swap_id -> Text,
+        reason -> Nullable<Text>,
+    }
 }
 
 table! {
-   rfc003_swaps {
-       id -> Integer,
-       swap_id -> Text,
-       role -> Text,
-       counterparty -> Text,
-   }
+    rfc003_swaps {
+        id -> Integer,
+        swap_id -> Text,
+        role -> Text,
+        counterparty -> Text,
+    }
+}
+
+// The new split protocol tables i.e., Han, HErc20, HALight.
+//
+
+table! {
+    created_swaps {
+        id -> Integer,
+        local_swap_id -> Text,
+        role -> Text,
+    }
+}
+
+// This is used by Alice to save dial info for created swaps.
+table! {
+    dial_infos {
+        id -> Integer,
+        local_swap_id -> Text,
+        dial_information -> Text,
+    }
+}
+// Alice can write this soon as she creates the swap, Bob can write it
+// after/during the communication protocol.
+table! {
+    secret_hashes {
+        id -> Integer,
+        local_swap_id -> Text,
+        secret_hash -> Text,
+    }
+}
+
+table! {
+    han_created_swaps {
+        id -> Integer,
+        local_swap_id -> Text,
+        network -> Text,            // i.e., 'regtest' for Bitcoin or '17' for Ethereum.  FIXME: Is this ok?
+        identity -> Text,
+        expiry -> BigInt,
+        amount -> Text,
+    }
+}
+
+table! {
+    herc20_created_swaps {
+        id -> Integer,
+        local_swap_id -> Text,
+        network -> Text,            // i.e., 'regtest' for Bitcoin or '17' for Ethereum.  FIXME: Is this ok?
+        identity -> Text,
+        expiry -> BigInt,
+        amount -> Text,
+        token_contract -> Text,
+    }
+}
+
+table! {
+    halight_created_swaps {
+        id -> Integer,
+        local_swap_id -> Text,
+        network -> Text,            // i.e., 'regtest' for Bitcoin or '17' for Ethereum.  FIXME: Is this ok?
+        identity -> Text,
```

Comment:
Should we make this explicit?

```suggestion
        my_identity -> Text,
```

Reply:
Yes, nice.  When done with @D4nte's suggestion of using `counterparty_alpha_identity` in finalized_swaps table, this works well.